tests/20-SideMenu/PinnedMenu.test.tsx

```diff
@@ -25,81 +28,216 @@ describe("Boards", () => {
+    render(<Icons />);
+    const actionReturn = jest.fn();
+    mocked(action).mockReturnValue(actionReturn);
+
+    const argsItems = ((Icons.args as WithPinnedSectionProps)
+    .items as (CircleButtonProps & { id: string, menuOptions?: DropdownProps["options"] })[]);
+
+    const linkArgs = argsItems.filter(item => !item.menuOptions);
+    
+    const iconLinks = screen.getAllByRole("link");
+
+    for (const [i, iconLink] of iconLinks.entries()) {
+      userEvent.click(iconLink);
+      await waitFor(() => {
+        expect(action).toBeCalledWith(`#${linkArgs[i].id}`);
+        expect(iconLink).toHaveAttribute("href", `#${linkArgs[i].id}`);
+      });
+    }
   });
   
+  test("Correctly marks icon without notification", async () => {
+    render(<Icons />);
+  
+    expect(screen.getByRole("link", { name: "heart"})).toBeInTheDocument; 
+    expect(screen.queryByLabelText("heart has new updates")).not.toBeInTheDocument; 
+    expect(document.querySelector(".has-dot")).not.toContainElement(screen.getByRole("link", { name: "heart"}));
+  });
+
+  test("Correctly marks icon with notification", async () => {
+    render(<Icons />);
+    
+    expect(within(document.querySelector(".has-dot")!).getByRole("link")).toHaveAccessibleName("inbox has new updates");
+  });
+
   test("Correctly marks current icon", async () => {
     render(<Icons />);
   
-    //TODO: fill this
+    const currentIcon = screen.getByRole("link", { current: "page" });
+    expect(currentIcon).toHaveAccessibleName("star");
+    expect(document.querySelector(".circle-button")).toContainElement(currentIcon);
+    expect(within(document.querySelector(".circle-button")!).getByRole("presentation")).toBeVisible();
   });
 
-  //TODO: Need to implement DropdownListMenu in PinnedMenu.tsx before testing this
   test("Renders icon options dropdown", async () => {
     render(<Icons />);
-  
-    //TODO: fill this
+    
+    const actionReturn = jest.fn();
+    mocked(action).mockReturnValue(actionReturn);
+    
+    userEvent.click(screen.getByLabelText("user options"));
+    await waitFor(() => {
+      expect(screen.getByText("Option 1")).toBeVisible();
+    });
+    userEvent.click(screen.getByText("Option 1"));    
+    await waitFor(() => {
+      expect(action).toBeCalledWith("userOption1");
+    });
+
+    await waitFor(() => {
+      expect(screen.queryByText("Option 1")).not.toBeInTheDocument();
+    })
+
+    userEvent.click(screen.getByLabelText("user options"));
+    await waitFor(() => {
+      expect(screen.getByText("Option 2")).toBeVisible();
+    });
+    userEvent.click(screen.getByText("Option 2"));    
+    await waitFor(() => {
+      expect(action).toBeCalledWith("userOption2");
+    });
   });
 });
 
 describe("Multiple Sections", () => {
   test("Renders sections in order", async () => {
     render(<MultipleSections />);
   
-    //TODO: fill this
+    const sections = document.getElementsByTagName("section");
```

Comment:
Would add a check on the length of sections here

Reply:
Will do